python/update_package_properties.py

```diff
@@ -0,0 +1,301 @@
+
+    return rpm_changelog
+
+
+def convert_citus_changelog_into_rpm_changelog(project_name: str, project_version: str, microsoft_email: str,
+                                               name_surname: str, fancy: bool, fancy_version_number: int,
+                                               changelog_date: date) -> str:
+    header = get_rpm_header(project_name, microsoft_email, name_surname, project_version, fancy,
+                            fancy_version_number, changelog_date)
+    rpm_changelog = f"{header.strip()}\n- Update to {project_name.capitalize()} {project_version}"
+
+    return rpm_changelog
+
+
+def update_rpm_spec(project_name: str, project_version: str, microsoft_email: str,
+                    name_surname: str, fancy: bool, fancy_version_number: int, spec_file_name: str,
+                    changelog_date: date, templates_path: str) -> None:
+    env = get_template_environment(templates_path)
+
+    fancy_version_str = get_version_number(project_version, fancy, fancy_version_number)
+    template = env.get_template('project.spec.tmpl')
+    rpm_changelog_history = get_rpm_changelog_history(project_name, spec_file_name)
+
+    history_lines = rpm_changelog_history.splitlines()
+
+    if len(history_lines) > 0 and project_version in history_lines[1]:
+        raise ValueError(f"{project_version} already exists in rpm spec file")
+
+    latest_changelog = convert_citus_changelog_into_rpm_changelog(project_name, project_version, microsoft_email,
+                                                                  name_surname, fancy, fancy_version_number,
+                                                                  changelog_date)
+    changelog = f"{latest_changelog}\n\n{rpm_changelog_history}"
+    content = template.render(version=project_version, fancy_version_str=fancy_version_str,
+                              fancy_version_no=fancy_version_number, changelog=changelog)
+    with open(spec_file_name, "w+") as writer:
+        writer.write(content)
+
+
+@validate_parameters
+def update_all_changes(github_token: non_empty(non_blank(str)), project_name: non_empty(str),
+                       project_version: is_version(str),
+                       tag_name: is_tag(str), fancy: bool, fancy_version_number: non_negative(int),
+                       microsoft_email: is_email(str),
+                       name_surname: non_empty(non_blank(str)), release_date: date, packaging_path: str):
+    templates_path = f"{BASE_PATH}/templates"
+    update_pkgvars(project_version, fancy, fancy_version_number, templates_path, f"{packaging_path}")
+    latest_changelog = get_changelog_for_tag(github_token, project_name, tag_name)
+    prepend_latest_changelog_into_debian_changelog(latest_changelog, project_version, fancy, fancy_version_number,
+                                                   f"{packaging_path}/debian/changelog", microsoft_email, name_surname,
+                                                   release_date)
```

Comment:
I think it would help readability and maintainability of this code if functions didn't have so many arguments. I think there's two main ways to solve this:
1. have all these arguments as global constants.
2. Create a class, put most of these on there as instance variables and make all these functions methods of this class.

I think 1 is the simplest to implement, but I think it might make it hard to test the functions then, since these globals would need to be set by the tests. The class approach is a bit more work, but probably keeps testing easy.

Reply:
Added ChangelogParameters class and changed method parameters into this class object

Reply:
Fixed
https://github.com/citusdata/tools/commit/be5591732f65aef151977d4523eb5ba560f5d5fd